src/SerialLoops/Dialogs/RomHacksDialog.cs

```diff
@@ -0,0 +1,217 @@
+                {
+                    overlays.Add(new(file, romInfoPath));
+                }
+
+                // Patch the overlays
+                string overlaySourceDir = Path.Combine(project.BaseDirectory, "src", "overlays");
+                for (int i = 0; i < overlays.Count; i++)
+                {
+                    if (Directory.GetDirectories(overlaySourceDir).Contains(Path.Combine(overlaySourceDir, overlays[i].Name)))
+                    {
+                        // If the overlay directory is empty, we've reverted all the hacks in it and should clean it up
+                        if (!Directory.GetFiles(Path.Combine(overlaySourceDir, overlays[i].Name, "source")).Any())
+                        {
+                            Directory.Delete(Path.Combine(overlaySourceDir, overlays[i].Name), true);
+                        }
+                        else
+                        {
+                            OverlayAsmHack.Insert(overlaySourceDir, overlays[i], newRomInfoPath, config.UseDocker ? config.DevkitArmDockerTag : string.Empty,
+                                (object sender, DataReceivedEventArgs e) => log.Log(e.Data),
+                                (object sender, DataReceivedEventArgs e) => log.LogWarning(e.Data));
+                        }
+                    }
+                }
+
+                // Save all the overlays in case we've reverted all hacks on one
+                foreach (Overlay overlay in overlays)
+                {
+                    overlay.Save(Path.Combine(project.BaseDirectory, "rom", "overlay", $"{overlay.Name}.bin"));
+                    File.Copy(Path.Combine(project.BaseDirectory, "rom", "overlay", $"{overlay.Name}.bin"),
+                        Path.Combine(project.IterativeDirectory, "rom", "overlay", $"{overlay.Name}.bin"), true);
+                    project.Settings.File.RomInfo.ARM9Ovt.First(o => o.Id == overlay.Id).RamSize = (uint)overlay.Length;
+                }
+
+                // We don't provide visible errors during the compilation of the hacks because it will deadlock the threads
+                // So at the end, we should check if any of the hacks that were supposed to be applied are not applied,
+                // and if there are some then we should let the user know.
+                IEnumerable<string> failedHackNames = appliedHacks.Where(h => !h.Applied(project)).Select(h => h.Name);
+                if (failedHackNames.Any())
+                {
+                    log.LogError($"Some hacks ({string.Join(", ", failedHackNames)}) failed to apply -- check logs to see why.");
+                }
+
+                Close();
+            };
+
+            StackLayout buttonsLayout = new()
+            {
+                Orientation = Orientation.Horizontal,
+                Spacing = 3,
+                Items =
+                {
+                    importButton,
+                    cancelButton,
+                    saveButton,
```

Comment:
```suggestion
                    saveButton,
                    cancelButton,
```
Save and cancel are the wrong way round here -- just need swapping around.